pages/deployment/[uuid].tsx

```diff
@@ -187,7 +191,21 @@ export default function Deployment() {
   const deploymentTag = `${uiConfig?.registry.host}/${deployment.metadata.contacts.requester}/${tag}`
 
   const requestApprovalReset = async () => {
-    await postEndpoint(`/api/v1/deployment/${deployment?.uuid}/reset-approvals`, {}).then((res) => res.json())
+    const response = await postEndpoint(`/api/v1/deployment/${deployment?.uuid}/reset-approvals`, {})
+
+    if (response.ok) {
+      setResetSnackbarStatus('success')
+      setResetSnackbarMessage('Approvals reset')
+      mutateDeployment()
+    } else {
+      setResetSnackbarStatus('error')
+      setResetSnackbarMessage(await getErrorMessage(response))
```

Comment:
As we're `await`ing during the second state update, we could end up in a situation where the status is out of sync with the message for a brief period of time. Could be written like so to avoid this:
```
      const errorMessage = await getErrorMessage(response)
      setResetSnackbarStatus('error')
      setResetSnackbarMessage(errorMessage)
```